packages/abi-coder/src/ResolvedAbiType.ts

```diff
@@ -168,6 +177,13 @@ export class ResolvedAbiType {
       return this.type;
     }
 
+    if (this.type.length > 1000) {
```

Comment:
What is the reasoning behind `1000`? We should probably take validation values from the [spec](https://docs.fuel.network/docs/specs/abi/fn-selector-encoding/).

Reply:
Another one resolved without the reasoning.

Also, wasn't this the perfect use case for named constants/variables?

```ts
    if (this.type.length > myMeaningfulVariableName) {
```

Just trying to understand where all those requests usually come from.